lib/dpl/provider/azure_webapps.rb

```diff
@@ -0,0 +1,40 @@
+module DPL
+  class Provider
+    class AzureWebApps < Provider
+      def config
+        {
+          "username"     => options[:username] || context.env['AZURE_WA_USERNAME'],
+          "password"     => options[:password] || context.env['AZURE_WA_PASSWORD'],
+          "site"         => options[:site] || context.env['AZURE_WA_SITE']
+        }
+      end
+
+      def git_target
+        "https://#{config['username']}:#{config['password']}@#{config['site']}.scm.azurewebsites.net:443/#{config['site']}.git"
```

Comment:
Since this is a `git`-based deployment, and if `git` authentication fails, it will spill the info:

```
fatal: Authentication failed for 'https://foo:bar@baz.scm.azurewebsites.net:443/baz.git/'
```

Could the git server on *.scm.azurewebsites.net does this for a wrong reason?